### PR TITLE
Centralize CI

### PR DESCRIPTION
**Pull request by sed-i**
## Issue
CI is outdated ([fails](https://github.com/canonical/traefik-k8s-operator/actions/runs/3594529878/jobs/6057922843)).

## Solution
Use centralized CI like all other o11y repos.

## Context
NTA.

## Testing Instructions
Make sure all tox envs are reflected in CI.

## Release Notes
Centralize CI.

### CODE REVIEW

**Comment by sed-i** [comment]
@PietroPasotti I may need your help in resolving the typing errors for the new `static-lib` and `static-charm` envs.

**Review by PietroPasotti — APPROVED** [review]
Noice.

File: lib/charms/traefik_k8s/v1/ingress.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent):
 # Provider ingress data model.
 ProviderIngressData = TypedDict("ProviderIngressData", {"url": str})
 # Provider application databag model.
-ProviderApplicationData = TypedDict("ProviderApplicationData", {"ingress": ProviderIngressData})
+ProviderApplicationData = TypedDict("ProviderApplicationData", {"ingress": ProviderIngressData})  # type: ignore
```
Comment (sed-i) [inline lib/charms/traefik_k8s/v1/ingress.py:127]: Pyright is having trouble here:
```
error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)
```

**Review by Abuelodelanada — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -77,9 +77,15 @@ deps =
...
+description = Static code checking
 deps =
-    pyright
-    -r{toxinidir}/requirements.txt
-commands = pyright --pythonversion 3.5 {toxinidir}/lib/charms/traefik_k8s/v1/ingress_per_unit.py
+    # TODO: drop the "charm:" qualification from the line below when we drop py3.5 support.
+    charm: pyright
+    charm: -r{toxinidir}/requirements.txt
+commands =
+    charm: pyright --pythonversion 3.8 {[vars]src_path}
```
[the remark this answers is not in the document]
Comment (sed-i) [reply tox.ini:87]: As discussed, traefik it is not going to be deployed on machines.

**Pull request merged**